app/routes/_auth.login.tsx

```diff
@@ -80,6 +107,41 @@ export default function Component() {
   const actionData = useActionData<typeof action>();
   const navigation = useNavigation();
   const isPending = navigation.state === 'submitting' || navigation.state === 'loading';
+
+  const submit = useSubmit();
```

Comment:
This implementation is using `useSubmit` and `useState`, but it might be possible to use `clientAction` to avoid using these hooks and needing client-side state.